pkg/manager/impl/task_manager.go

```diff
@@ -120,6 +121,15 @@ func (t *TaskManager) CreateTask(
 			contextWithRuntimeMeta, common.RuntimeVersionKey, finalizedRequest.Spec.Template.Metadata.Runtime.Version)
 		t.metrics.Registered.Inc(contextWithRuntimeMeta)
 	}
+
+	if request.Spec.DescriptionEntity != nil {
+		err = createDescriptionEntity(ctx, t.db, request.Spec.DescriptionEntity, *request.Id)
```

Comment:
not atomic though... so we create the task first, then we try to create the description entity, and then we update the task/workflow with the description id.  i'd feel better if we can do this in one database transaction instead of three.

Also I'm a little confused here about what the flow is... for large descriptions, how/when does that get uploaded?  Is that something done client side by calling the data proxy for a signed url to upload?  I thought that offloading was going to happen in Admin and I'm not seeing it but I could be missing something, or maybe i'm remembering incorrectly.  how did we say the offloading process would work?

Reply:
if we're not making this atomic then can we consider making a separate endpoint for registering description entities?

Reply:
made this atomic